src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java

```diff
@@ -0,0 +1,394 @@
+package com.minecolonies.coremod.placementhandlers;
+
+import com.minecolonies.coremod.blocks.BlockSolidSubstitution;
+import com.minecolonies.coremod.configuration.Configurations;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIStructure;
+import com.minecolonies.coremod.util.BlockUtils;
+import com.minecolonies.coremod.util.InventoryUtils;
+import net.minecraft.block.*;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemDoor;
+import net.minecraft.item.ItemFlintAndSteel;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class containing all placement handler implementations.
+ */
+public final class PlacementHandlers
+{
+    public static final List<IPlacementHandler> handlers = new ArrayList<>();
+
+    private PlacementHandlers()
+    {
+        /**
+         * Intentionally left empty.
+         */
+    }
+
+    static
+    {
+        handlers.add(new AirPlacementHandler());
+        handlers.add(new FireplacementHandler());
+        handlers.add(new GrassPlacementHandler());
+        handlers.add(new DoorPlacementHandler());
+        handlers.add(new BedPlacementHandler());
+        handlers.add(new DoublePlantPlacementHandler());
+        handlers.add(new SpecialBlockPlacementAttemptHandler());
+        handlers.add(new FlowerPotPlacementHandler());
+        handlers.add(new BlockGrassPathPlacementHandler());
+        handlers.add(new StairBlockPlacementHandler());
+        handlers.add(new BlockSolidSubstitutionPlacementHandler());
+        handlers.add(new GeneralBlockPlacementHandler());
+    }
+
+    //If he woudln't count the bracket spaces we'd be under 25 easily.
+    public static class FireplacementHandler implements IPlacementHandler
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)